File: src/main/java/frc/robot/commands/elevator/SafeElevatorDown.java
package frc.robot.commands.elevator;

import edu.wpi.first.wpilibj.command.CommandGroup;
import frc.robot.subsystems.Elevator;
import frc.robot.subsystems.MouthArticulator;
import frc.robot.commands.cargo.ArticulateMouth;

/**
 * It has been determined the elevator is moving down and may collide with the cargo intake.
 * Therefore, we will move the intake and the elevator in such a way they don't do that.
 */
public class SafeElevatorDown extends CommandGroup {

    public SafeElevatorDown(Elevator.ElevatorPosition targetPosition) {
        addParallel(new AutomaticElevator(Elevator.ElevatorPosition.ROCKET_1_C, true));
        addSequential(new ArticulateMouth(MouthArticulator.MouthPosition.OUT));
        addSequential(new AutomaticElevator(targetPosition, true));
        addSequential(new ArticulateMouth(MouthArticulator.MouthPosition.IN));
    }
}

File: src/main/kotlin/ktfrc/robot/commands/elevator/SafeElevatorDown.kt
package ktfrc.robot.commands.elevator

import edu.wpi.first.wpilibj.command.CommandGroup
import ktfrc.robot.commands.cargo.ArticulateMouth
import ktfrc.robot.subsystems.ElevatorPosition
import ktfrc.robot.subsystems.MouthArticulatorPosition

class SafeElevatorDown(targetPosition: ElevatorPosition): CommandGroup() {
    init {
        addParallel(AutomaticElevator(ElevatorPosition.ROCKET_1_C, endable = true))
        addSequential(ArticulateMouth(MouthArticulatorPosition.OUT))
        addSequential(AutomaticElevator(targetPosition, endable = true))
        addSequential(ArticulateMouth(MouthArticulatorPosition.IN))
    }
}